spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/ChannelBindingService.java

```diff
@@ -56,90 +57,105 @@
 
 	private final Log log = LogFactory.getLog(ChannelBindingService.class);
 
-	private BinderFactory<MessageChannel> binderFactory;
+	private BinderFactory binderFactory;
 
 	private final ChannelBindingServiceProperties channelBindingServiceProperties;
 
-	private final Map<String, Binding<MessageChannel>> producerBindings = new HashMap<>();
+	private final Map<String, Binding<?>> producerBindings = new HashMap<>();
 
-	private final Map<String, List<Binding<MessageChannel>>> consumerBindings = new HashMap<>();
+	private final Map<String, List<Binding<?>>> consumerBindings = new HashMap<>();
 
-	public ChannelBindingService(ChannelBindingServiceProperties channelBindingServiceProperties,
-			BinderFactory<MessageChannel> binderFactory) {
+	public ChannelBindingService(
```

Comment:
+1 to Ilaya's suggestion (as well as removing all mentions of "channel" from the JavaDoc), but let's take the opportunity to rename the "Service" part as well. I'm thinking... will post back if I have some ideas.